experimental/Schemes/WeilDivisor.jl

```diff
@@ -133,6 +134,11 @@ function copy(D::WeilDivisor)
   return WeilDivisor(scheme(D), coefficient_ring(D), new_dict, check=false)
 end
 
+function irreducible_decomposition(D::WeilDivisor)
```

Comment:
cache this?

Reply:
definitely.

Reply:
I see the point. We are quite heavy on memory already.